test/Jsr292/src/com/ibm/j9/jsr292/indyn/IndyTest.java

```diff
@@ -211,17 +211,21 @@ public void test_indyn_call_Illegal_crosspackage_method () {
 	//Negative test : invalid bootstrap method
 	@Test(groups = { "level.extended" })
 	public void test_indyn_invalid_bootstrap_method() {
-		
-		boolean bootstrapMethodError = false;
-		
 		try {
 			String s = com.ibm.j9.jsr292.indyn.GenIndyn.invalid_bootstrap(new Object());
-		} catch ( BootstrapMethodError e ) {
-			bootstrapMethodError = true;
-		}
-		
-		if ( bootstrapMethodError == false ) {
-			Assert.fail("BootstrapMethodError not thrown when invalid bootstrap method is used");
+			Assert.fail("Exception should be thrown");
+		} catch (BootstrapMethodError e) {
+			//BootstrapMethodError should only apply to Java8
+			if (false == "1.8".equals(System.getProperty("java.specification.version"))) {
+				Assert.fail("Unexpected exception ", e);
```

Comment:
in all assert messages, we must highlight expected behavior. it will be easier to understand test behavior without reading test code.
```
	public void test_indyn_invalid_bootstrap_method() {
		try {
			String s = com.ibm.j9.jsr292.indyn.GenIndyn.invalid_bootstrap(new Object());
		} catch (BootstrapMethodError e) {
			if (false == "1.8".equals(System.getProperty("java.specification.version"))) {
				Assert.fail("[Java 9+] NoSuchMethodError not thrown when a non-existent bootstrap method is used");
			}
			return;
		} catch (NoSuchMethodError e) {
			if ("1.8".equals(System.getProperty("java.specification.version"))) {
				Assert.fail("[Java 8] BootstrapMethodError not thrown when invalid bootstrap method is used");
			}
			return;
		}
		Assert.fail("BootstrapMethodError (in Java 8) or NoSuchMethodError (Java 9+) should have been thrown");
	}
```

Reply:
expected behaviour is described in the comment

Reply:
didn't see `e` was parsed in the message

Reply:
sorry, misunderstand the suggestions, yes, we also need to put the exception in the failure message. PR updated.